### DISCUSSION TITLE
How to catch exception raise in Response.__call__?

### DISCUSSION THREAD

**Discussion by gshmu** [fastapi/fastapi#12078] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.responses import FileResponse

app = FastAPI()

@app.get("/download/")
async def download_file(filename: str):
    try:
        return FileResponse(f"/path/to/nonexistent/{filename}")
    except Exception as e:
        return {"error": str(e)}
```

### Description

Describe how FileResponse does not raise an exception that can be caught using a try/except block when a file is missing.

### Operating System

Linux

### FastAPI Version

0.112.1

### Pydantic Version

2.8.2

### Python Version

Python 3.12.4

### Additional Context

I can check `os.path.exists` for FileResponse, but I want find solution for all Response call err

**Comment by sinisaos** [comment]
@gshmu Sorry if I don't understand your request correctly, but `RuntimeError` is raised with or without a `try/except` block. `FileResponse` is a Starlette response and [here is](https://github.com/encode/starlette/blob/c78c9aac17a4d68e0647252310044502f1b7da71/starlette/responses.py#L327-L328) code for an exception if the file does not exist.

**Comment by YuriiMotov** [comment, marked as answer]
Haven't checked it, but I think you can create a subclass of `FileResponse` and define `__call__` method as:

```
class MyFileResponse(FileResponse):
    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
        try:
            await super().__call__(scope, receive, send)
        except RuntimeError as e:
            pass  # Process exception
```

Another way is to check whether the file path is correct before returning `FileResponse` (I would choose this way)
